### ISSUE TITLE
[Feature Request]:  Players who support offline mode use

### ISSUE THREAD

**Issue by QianMuu** [FTBTeam/FTB-Mods-Issues#785]
### Mod

FTB Teams

### Description

Players in offline mode cannot use it.

### Describe the solution you'd like

I hope that 'FTB Teams' can support players in both online and offline modes to use team functions normally

### Describe alternatives you've considered

FTB Teams "is a great module, its features are really great, and it has been very helpful for me to build servers.

However, if it is not possible to support equal use of team functions for both online and offline players.

I may consider using an online mode where the server is turned off, but there is no way.

I am somewhat disappointed that the Minecraft account I purchased cannot access the server I built.

I can't use my skin either.

**Comment by DJsushi** [comment]
Has there been any progress towards this? I'm curious as to why this limitation even exists. Could someone please explain it to me with some technical details?

**Comment by desht** [comment]
No, this will never happen.  The only reason to allow both online and offline mode players at the same time is to support players who haven't bought the game.

And we do not support software piracy in any way.

**Comment by DJsushi** [comment]
@desht but I wasn't asking about both online and offline players. I was asking about a purely offline server that runs on a LAN network without an internet connection. For some reason, it doesn't work for me, the data isn't received from the server and the message "Team data has not been received from the server! FTB Teams will not function correctly. Check server log for errors, and ensure your server is in online mode".

**Comment by desht** [comment]
Strange, since I just tested connecting to an offline server with my purchased account, and it works fine.

**Comment by DJsushi** [comment]
> Strange, since I just tested connecting to an offline server with my purchased account, and FTB Teams works fine.

@desht I don't know what's wrong. I use PrismLauncher (not the cracked version) to launch the game. I tried launching both in online and offline mode (and my local server is in offline mode) and it doesn't work either way. The funny thing is, that when I cloned the FTB Teams repo myself and built it using Gradle and then launched both a server and client, it worked. I don't know what can be causing this but it's messing a lot with my All of Fabric world.

**Comment by DJsushi** [comment]
Yes, the UUID is not known to Mojang, the server has only been running in offline mode and the connected client is also run in offline mode through the PrismLauncher.

**Comment by IceOfSummer** [comment]
It seems that offline mode players do not have an uuid：
```shell
# decomplied code: Minecraft.getInstance().getUser().getGameProfile().getId();
[arthas@44608]$ ognl '@net.minecraft.client.Minecraft@func_71410_x().func_110432_I().func_148256_e()' -c 6fe337a5
@GameProfile[
    id=null,
    name=@String[IceBing],
    properties=@PropertyMap[{}],
    legacy=@Boolean[false],
]

[arthas@44608]$ ognl '@dev.ftb.mods.ftbteams.data.ClientTeamManager@INSTANCE' -c 6fe337a5
@ClientTeamManager[
    INSTANCE=@ClientTeamManager[dev.ftb.mods.ftbteams.data.ClientTeamManager@2e48b57b],
    invalid=@Boolean[false],
    id=@UUID[b1f1f861-8490-455c-9b33-6f4fe843d991],
    teamMap=@HashMap[isEmpty=false;size=1],
    knownPlayers=@HashMap[isEmpty=false;size=1],
    selfTeam=@ClientTeam[IceBing#33bc027a],
    selfKnownPlayer=null,
]
```

Maybe this can be replaced by using the username to get the corresponding user?

**Comment by MichaelHillcox** [comment]
We have no intention of adding support to specifically support running offline, sorry.
